src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs

```
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.DotNet.Build.Tasks;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.GenFacades
+{
+    public class NotSupportedAssemblyGenerator : BuildTask
+    {
+        [Required]
+        public ITaskItem[] SourceFiles { get; set; }
+
+        [Required]
+        public string IntermediateOutputPath { get; set; }
+
+        [Required]
+        public string Message { get; set; }
+
+        public string ApiExclusionListPath { get; set; }
+
+        public override bool Execute()
+        {
+           GenerateNotSupportedAssemblyFiles(SourceFiles?.Select(item => item.ItemSpec).ToList(), IntermediateOutputPath, Message, ApiExclusionListPath);
```

NIT: Should we throw an error if SourceFiles is empty?